ssl/statem/statem_srvr.c

```diff
@@ -324,11 +332,17 @@ static int send_server_key_exchange(SSL *s)
  *   1: Yes
  *   0: No
  */
-static int send_certificate_request(SSL *s)
+int send_certificate_request(SSL *s)
 {
     if (
            /* don't request cert unless asked for it: */
            s->verify_mode & SSL_VERIFY_PEER
+           /*
+            * don't request if post-handshake-only unless doing
+            * post-handshake in TLSv1.3
+            */
+           && (!SSL_IS_TLS13(s) || !(s->verify_mode & SSL_VERIFY_POST_HANDSHAKE)
+               || s->post_handshake_auth == SSL_PHA_REQUEST_PENDING)
            /*
             * if SSL_VERIFY_CLIENT_ONCE is set, don't request cert
```

Comment:
`SSL_VERIFY_CLIENT_ONCE`, effectively disables post-handshake authentication; since it prevents a certificate request from happening during a renegotiation (anything after the initial handshake). To make it work with post-handshake authentication, I'm thinking a certificate received flag/counter might be needed.

Reply:
So, I ever-so-slightly changed the behavior of SSL_VERIFY_CLIENT_ONCE to only send a CertReq once (period). This behavior is mostly consistent with the existing behavior; especially since the s_server.c code sets the SSL_VERIFY_CLIENT_ONCE flag during cert renegotiation (when it receives `GET /renegcert` which shouldn't work according to the documentation). That means that post-handshake auth will fail immediately if certificate authentication previously occurred during the initial handshake, or will fail the second time if certificate authentication did not occur during the initial handshake.